src/sentry/static/sentry/app/views/settings/components/forms/projectMapperField.tsx

```diff
@@ -0,0 +1,167 @@
+    };
+
+    const renderItem = (itemTuple: [number, any]) => {
+      const [projectId, mappedValue] = itemTuple;
+      const {slug} = sentryProjectsById[projectId];
+      const {label: itemLabel} = mappedItemsByValue[mappedValue];
+      return (
+        <Item key={projectId}>
+          <ItemValue>{slug}</ItemValue> <ItemValue>{itemLabel}</ItemValue>
+        </Item>
+      );
+    };
+
+    const customValueContainer = containerProps => {
+      const valueList = containerProps.getValue();
+      //if no value set, we want to return the default component that is rendered
+      if (valueList.length === 0) {
+        return <components.ValueContainer {...containerProps} />;
+      }
+      const projectId = valueList[0].value;
+      const project = sentryProjectsById[projectId];
+      if (!project) {
+        return <components.ValueContainer {...containerProps} />;
+      }
+      return (
+        <components.ValueContainer {...containerProps}>
+          <IdBadge
+            project={project}
+            avatarSize={20}
+            displayName={project.slug}
+            avatarProps={{consistentWidth: true}}
+          />
+        </components.ValueContainer>
+      );
+    };
+
+    const customOptionProject = projectProps => {
+      const project = sentryProjectsById[projectProps.value];
+      if (!project) {
+        return null;
+      }
+      return (
+        <components.Option {...projectProps}>
+          <IdBadge
+            project={project}
+            avatarSize={20}
+            displayName={project.slug}
+            avatarProps={{consistentWidth: true}}
+          />
+        </components.Option>
+      );
+    };
+
+    return (
+      <Wrapper>
+        {existingValues.map(renderItem)}
+        <SelectContainer>
+          <StyledSelectControl
+            placeholder={t('Select a Project')}
+            name="project"
+            openMenuOnFocus
+            options={projectOptions}
+            components={{
+              Option: customOptionProject,
+              ValueContainer: customValueContainer,
+            }}
+            ref={this.sentryProjectRef}
+          />
+          <StyledSelectControl
+            placeholder={mappedDropdown.placeholder}
+            name="mappedDropwdown"
+            openMenuOnFocus
+            options={mappedItemsToShow}
+            ref={this.mappedRef}
+          />
+          <Button size="small" label={t('Add')} icon={<IconAdd />} onClick={handleAdd} />
+        </SelectContainer>
+      </Wrapper>
+    );
+  }
+}
+
+const ProjectMapperField = (props: Props) => (
+  <InputField
+    {...props}
+    field={(renderProps: RenderProps) => <RenderField {...renderProps} />}
+  />
+);
+
+export default ProjectMapperField;
+
+const StyledSelectControl = styled(SelectControl)`
+  width: 50%;
+`;
+
+const SelectContainer = styled('div')`
+  display: flex;
+`;
+
+const Wrapper = styled('div')``;
+
+const Item = styled('div')`
+  padding: 5px;
```

Comment:
Can we use `space`?

Reply:
@EvanPurkhiser All the styles are going to change in a future PR when I get the design from Chris